src/javascript/extendable_audio.js

```diff
@@ -0,0 +1,46 @@
+/// <reference path="./extendable_audio.d.ts" />
+
+const SKIP_PROPERTIES = ['dispatchEvent', 'addEventListener', 'removeEventListener'];
+
+export class ExtendableAudio extends DocumentFragment {
+  constructor(url) {
+    this._audio = new Audio();
+    this._documentFragment = document.createDocumentFragment();
+    this._toSetUrl = url;
+    var proto = Object.getPrototypeOf(this);
+    for (let property in this._audio) {
+      if (SKIP_PROPERTIES.indexOf(property) == -1 &&
+        !(proto.hasOwnProperty(property) || this.hasOwnProperty(property))) {
+        if (typeof this._audio[property] === 'function') {
+          Object.defineProperty(this, property, {
+            enumerable: false,
+            value: this._audio[property].bind(this._audio),
+            writable: false
+          });
+        } else {
+          Object.defineProperty(this, property, {
+            get: () => this._audio[property],
+            set: (value: any)  => this._audio[property] = value
+          });
+        }
+      }
+    }
+  }
```

Comment:
Basically just looping through the properties of the Audio() element and proxying through to them. We could have used a real ES6 proxy here, but that would have been less compatible and introduced some other problems.